### DISCUSSION TITLE
Deploying GLPI-Agent on all workstations

### DISCUSSION THREAD

**Discussion by sebma** [glpi-project/glpi-agent#284] [Q&A] [answered]
Hi, I've read [here](../../glpi-agent#description) that GLPI Agent is compatible with FusionInventory plugin.

To prepare the migration of our GLPI v9.5.7+FusionInventory plugin 9.5+4.2 to GLPI v10, I need to know how to deploy GLPI-Agent on all workstations with the help of the already installed FusionInventory Agent v2.6 ?

**Comment by g-bougard** [comment, marked as answer]
Hi @sebma 
do you mean you want to run the Deploy task from FusionInventory agents to replace it by GLPI agents ?
If yes, I can suggest you to first create a Deploy package to install GLPI agent silently, check [Windows installer documentation](https://glpi-agent.readthedocs.io/en/latest/installation/windows-command-line.html). This way you'll finally have 2 agents running on each computers: this is not a problem, the only caveat is the agent http port (62354) won't be shared and will be kept by FI agent until you uninstall it and restart GLPI agent. During the migration, not having access to this port for GLPI Agent is not critical. After GLPI agents are installed, you can create a Deploy package to uninstall FI agents and run this package only on GLPI agents. In this package, you can also schedule a task which will restart the agent after a short time to it can finally listen on dedicated httpd port.
Another way is to use the vbscript (see [Large installations](https://glpi-agent.readthedocs.io/en/latest/installation/index.html#large-installations). This script should be configured to your need (options to update in the first lines). It is generally used with GPO, but it's not forbidden to use it from Deploy task. This script has an option to uninstall FI agent but if you use it, you must configure your package to run the vbscript in background after a few time to leave the task finish in FI agent before it is uninstalled otherwise the task will probably be leaved in a running or failed status in FI plugin and even this will probably abort the installation.
Hope this help

**Comment by sebma** [comment]
Hi @g-bougard Thanks a lot for your explanation :+1: .

**Reply by keytrap-x86** [reply]
Hello, if you need extra informations please see my comment https://github.com/glpi-project/glpi-agent/discussions/272#discussioncomment-4215226
